pkg/provider-local/webhook/controlplane/ensurer.go

```diff
@@ -122,3 +116,52 @@ func containsFilePath(files *[]extensionsv1alpha1.File, filePath string) bool {
 
 	return false
 }
+
+func (e *ensurer) EnsureCRIConfig(_ context.Context, _ extensionscontextwebhook.GardenContext, new, _ *extensionsv1alpha1.CRIConfig) error {
+	if !extensionsv1alpha1helper.HasContainerdConfiguration(new) {
+		return nil
+	}
+
+	for _, registry := range []extensionsv1alpha1.RegistryConfig{
+		{
+			Upstream: "gcr.io",
+			Server:   ptr.To("https://gcr.io"),
+			Hosts:    []extensionsv1alpha1.RegistryHost{{URL: "http://garden.local.gardener.cloud:5003"}},
+		},
+		{
+			Upstream: "eu.gcr.io",
+			Server:   ptr.To("https://eu.gcr.io"),
+			Hosts:    []extensionsv1alpha1.RegistryHost{{URL: "http://garden.local.gardener.cloud:5004"}},
+		},
+		{
+			Upstream: "ghcr.io",
+			Server:   ptr.To("https://ghcr.io"),
+			Hosts:    []extensionsv1alpha1.RegistryHost{{URL: "http://garden.local.gardener.cloud:5005"}},
+		},
+		{
+			Upstream: "registry.k8s.io",
+			Server:   ptr.To("https://registry.k8s.io"),
+			Hosts:    []extensionsv1alpha1.RegistryHost{{URL: "http://garden.local.gardener.cloud:5006"}},
+		},
+		{
+			Upstream: "quay.io",
+			Server:   ptr.To("https://quay.io"),
+			Hosts:    []extensionsv1alpha1.RegistryHost{{URL: "http://garden.local.gardener.cloud:5007"}},
+		},
+		{
+			Upstream: "europe-docker.pkg.dev",
+			Server:   ptr.To("https://europe-docker.pkg.dev"),
+			Hosts:    []extensionsv1alpha1.RegistryHost{{URL: "http://garden.local.gardener.cloud:5008"}},
+		},
+	} {
+		addRegistryIfNotAvailable(registry, new.Containerd)
+	}
+
+	return nil
+}
+
+func addRegistryIfNotAvailable(registry extensionsv1alpha1.RegistryConfig, config *extensionsv1alpha1.ContainerdConfig) {
```

Comment:
As we talked offline, let's add a comment that explains the reasoning behind the "if not avaiable" handling. Just to prevent someone from changing it to "add or update" handling in the future.